packages/material-ui/src/Typography/Typography.js

```diff
@@ -1,106 +1,59 @@
-  /* Styles applied to the root element if `variant="overline"`. */
-  overline: theme.typography.overline,
-  /* Styles applied to the root element if `variant="inherit"`. */
-  inherit: {},
-  /* Styles applied to the root element if `align="left"`. */
-  alignLeft: {
-    textAlign: 'left',
-  },
-  /* Styles applied to the root element if `align="center"`. */
-  alignCenter: {
-    textAlign: 'center',
-  },
-  /* Styles applied to the root element if `align="right"`. */
-  alignRight: {
-    textAlign: 'right',
-  },
-  /* Styles applied to the root element if `align="justify"`. */
-  alignJustify: {
-    textAlign: 'justify',
-  },
-  /* Styles applied to the root element if `nowrap={true}`. */
-  noWrap: {
+const getTextColor = (color, palette) => {
+  if (color.indexOf('text') === 0) {
+    return palette.text[color.split('text').pop().toLowerCase()];
+  }
+
+  if (color === 'inherit' || color === 'initial') {
+    return color;
+  }
+
+  return palette[color].main;
+};
+
+const overridesResolver = (props, styles) => {
+  const styleOverrides = {
+    ...styles.root,
+  }
+
+  return styleOverrides;
+};
+
+export const TypographyRoot = experimentalStyled(
+  'span',
+  {},
+  { muiName: 'MuiTypography', overridesResolver },
+)((props) => ({
+  margin: 0,
+  ...(props.styleProps.variant && props.theme.typography[props.styleProps.variant]),
+  ...(props.styleProps.align !== 'inherit' && {
+    textAlign: props.styleProps.align,
+  }),
+  ...(props.styleProps.noWrap && {
     overflow: 'hidden',
     textOverflow: 'ellipsis',
     whiteSpace: 'nowrap',
-  },
-  /* Styles applied to the root element if `gutterBottom={true}`. */
-  gutterBottom: {
+  }),
+  ...(props.styleProps.gutterBottom && {
     marginBottom: '0.35em',
-  },
-  /* Styles applied to the root element if `paragraph={true}`. */
-  paragraph: {
+  }),
+  ...(props.styleProps.paragraph && {
     marginBottom: 16,
-  },
-  /* Styles applied to the root element if `color="inherit"`. */
-  colorInherit: {
-    color: 'inherit',
-  },
-  /* Styles applied to the root element if `color="primary"`. */
-  colorPrimary: {
-    color: theme.palette.primary.main,
-  },
-  /* Styles applied to the root element if `color="secondary"`. */
-  colorSecondary: {
-    color: theme.palette.secondary.main,
-  },
-  /* Styles applied to the root element if `color="textPrimary"`. */
-  colorTextPrimary: {
-    color: theme.palette.text.primary,
-  },
-  /* Styles applied to the root element if `color="textSecondary"`. */
-  colorTextSecondary: {
-    color: theme.palette.text.secondary,
-  },
-  /* Styles applied to the root element if `color="error"`. */
-  colorError: {
-    color: theme.palette.error.main,
-  },
-  /* Styles applied to the root element if `display="inline"`. */
-  displayInline: {
-    display: 'inline',
-  },
-  /* Styles applied to the root element if `display="block"`. */
-  displayBlock: {
-    display: 'block',
-  },
-});
+  }),
+  ...(props.styleProps.color && {
```

Comment:
```suggestion
  ...(props.styleProps.color && props.styleProps.color !== 'initial' && {
```

See https://github.com/mui-org/material-ui/pull/23841#issuecomment-740666993 for details